mix.exs

```diff
@@ -54,4 +55,30 @@ defmodule Shoehorn.Mixfile do
       links: %{"GitHub" => "https://github.com/nerves-project/shoehorn"}
     ]
   end
+
+  def enforce_handler() do
+    unless Application.get_env(:shoehorn, :handler) do
+      Mix.raise("""
+      The default strategy for how Shoehorn handles OTP application exits has changed.
+      Before this release, if an application were to exit, the node would remain running
+      and that applications would remain stopped. This may be desirable for development
+      and test but is typically undesirable in production. This behavior can be
+      customized by configuring the `handler` in the config. For example, in dev you can
+      use the module `Shoehorn.Handler.Ignore` to prevent the node from halting on failure. 
+
+        # config/dev.exs
+
+        config :shoehorn,
+          handler: Shoehorn.Handler.Ignore
+
+      In prod, you can specify the default handler, or implement your own.
+
+        # config/dev.exs
+
+        config :shoehorn,
+          handler: Shoehorn.Handler.Default
```

Comment:
I realize that I didn't appreciate the naming of `Default` in previous reviews, but now that I've been burned by it, can it be renamed to `Shoehorn.Handler.Halt` (since it calls `:erlang.halt`) or something else? Plus it doesn't feel like a default since you have to specify it or you get this error.